### PR TITLE
Add a test for passing a struct from a const array

### PR DESCRIPTION
**Pull request by Oletus**
This covers bug http://crbug.com/871434 . The root cause seems to be
in the NVIDIA OpenGL driver.

### CODE REVIEW

File: sdk/tests/conformance2/glsl3/const-struct-from-array-as-function-parameter.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+// Covers bug:
+// http://crbug.com/871434
+
+GLSLConformanceTester.runTests([
+{
+  fShaderId: 'fShaderTest',
+  fShaderSuccess: true,
+  linkSuccess: true,
+  passMsg: 'Passing a struct from a dynamically indexed const array as a non-const function parameter should work.',
```
Comment (jdashg) [inline sdk/tests/conformance2/glsl3/const-struct-from-array-as-function-parameter.html]: s/as/into/
The spec VM treats structs passed into functions as copies, but 'as' sounds like it's a reference. While under the covers, the copy is certainly possible to optimize out, from the spec's standpoint, structs are always copied in. (and out, when marked as such)

Comment (Oletus) [reply sdk/tests/conformance2/glsl3/const-struct-from-array-as-function-parameter.html]: Done.

**Comment by jdashg** [comment]
Thanks!

File: sdk/tests/conformance2/glsl3/const-struct-from-array-as-function-parameter.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+bool useStruct( S s) { return s.member > 0.0; }
+
+out vec4 outColor;
+
+void main( void )
+{
+    outColor = vec4(0.0, 0.0, 0.0, 1.0);
+    for (int i = 0; i < 2; ++i) {
+        if (useStruct(s[i])) {
+            outColor.g++;
```
Comment (kenrussell) [inline sdk/tests/conformance2/glsl3/const-struct-from-array-as-function-parameter.html]: Is it intended in this test that g can evaluate to a value > 1.0?

**Pull request merged**